src/libs/auth.js

```diff
@@ -298,9 +295,9 @@ authStore.subscribe(withErrorReporting('Error checking registration', async (sta
     try {
       const { enabled } = await Ajax().User.getStatus()
       if (enabled) {
-        // While initial state is first loading, state.termsOfService.userCanUseTerra will be undefined (it will then be `true` on the
-        // second execution of this code, which is still part of the initial rendering).
-        return state.termsOfService.userCanUseTerra ? userStatus.registeredWithTos : userStatus.registeredWithoutTos
+        // While initial state is first loading, state.termsOfService.acceptedTosAllowsUsage will be undefined
+        // (it will then be `true` on the second execution of this code, which is still part of the initial rendering).
```

Comment:
I'm also confused... this code is part of global state management, which doesn't really have anything to do with rendering (at least "rendering" in the React sense).

I think it might be clearer to say something like "When Terra is first loaded, termsOfService.permitsSystemUsage will be undefined while the user's ToS status is fetched from Sam".

Reply:
updated comment